artwork/Modules/InventoryManagement/Repositories/CraftInventoryCategoryRepository.php

```diff
@@ -5,32 +5,47 @@
 use Artwork\Core\Database\Repository\BaseRepository;
 use Artwork\Modules\InventoryManagement\Models\CraftInventoryCategory;
 use Illuminate\Database\Eloquent\Collection;
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Database\Query\Builder as BaseBuilder;
 
-readonly class CraftInventoryCategoryRepository extends BaseRepository
+class CraftInventoryCategoryRepository extends BaseRepository
 {
-    public function __construct()
+    public function __construct(private readonly CraftInventoryCategory $craftInventoryCategory)
     {
     }
 
-    public function find(int $id): CraftInventoryCategory
+    public function getNewModelInstance(array $attributes = []): CraftInventoryCategory
     {
-        /** @var CraftInventoryCategory $craftInventoryCategory */
-        $craftInventoryCategory = CraftInventoryCategory::find($id);
+        return $this->craftInventoryCategory->newInstance($attributes);
+    }
+
+    public function getNewModelQuery(): BaseBuilder|Builder
+    {
+        /** @var BaseBuilder|Builder $builder */
+        $builder = $this->craftInventoryCategory->newModelQuery();
+
+        return $builder;
+    }
+
+    public function find(int $id): CraftInventoryCategory|null
+    {
+        /** @var CraftInventoryCategory|null $craftInventoryCategory */
+        $craftInventoryCategory = $this->getNewModelQuery()->find($id);
```

Comment:
```suggestion
        return $this->getNewModelQuery()->find($id);
```

Reply:
Das ist absichtlich so damit PHPStorm die Objekte richtig resolved und man in der Vervollständigung z.B. beim Model auch direkt alle relations hat und man nicht immer rumraten muss. 👍